azext_edge/edge/params.py

```diff
@@ -302,6 +310,822 @@ def load_iotops_arguments(self, _):
+            "secret_name",
+            options_list=["--secret-name", "--sn"],
+            help="The name of the Kubernetes secret that contains the SASL token.",
+            arg_group="SASL Authentication",
+        )
+        context.argument(
+            "authentication_type",
+            options_list=["--auth-type"],
+            choices=CaseInsensitiveList(
+                [
+                    DataflowEndpointAuthenticationType.SYSTEMASSIGNED.value,
+                    DataflowEndpointAuthenticationType.USERASSIGNED.value,
+                    DataflowEndpointAuthenticationType.SASL.value,
+                ]
+            ),
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create localstorage") as context:
+        context.argument(
+            "pvc_reference",
+            options_list=["--pvc-ref"],
+            help="The name of the PersistentVolumeClaim (PVC) to use for local storage. "
+            "Note: The PVC must be in the same namespace as the data flow endpoint.",
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create local-mqtt") as context:
+        context.argument(
+            "hostname",
+            options_list=["--hostname"],
+            help="The hostname of the local MQTT broker.",
+        )
+        context.argument(
+            "port",
+            options_list=["--port"],
+            help="The port number of the local MQTT broker.",
+            type=int,
+        )
+        context.argument(
+            "audience",
+            options_list=["--audience", "--aud"],
+            help="The audience of the Kubernetes service account token (SAT).",
+            arg_group="Kubernetes Service Account Token",
+        )
+        context.argument(
+            "secret_name",
+            options_list=["--secret-name", "--sn"],
+            help="The name for the kubernetes secret that contains the X509 client certificate, private key "
+            "corresponding to the client certificate, and intermediate certificates for the client certificate "
+            "chain. "
+            "Note: The certificate and private key must be in PEM format and not password protected.",
+            arg_group="X509 Authentication",
+        )
+        context.argument(
+            "authentication_type",
+            options_list=["--auth-type"],
+            choices=CaseInsensitiveList(
+                [
+                    DataflowEndpointAuthenticationType.SERVICEACCESSTOKEN.value,
+                    DataflowEndpointAuthenticationType.X509.value,
+                ]
+            ),
+        )
+
+    with self.argument_context("iot ops dataflow endpoint create eventgrid") as context:
+        context.argument(
+            "hostname",
+            options_list=["--hostname"],
+            help="The hostname of the event grid namespace. Can"
+            " be found in 'Http hostname' property. In the form "
+            "of NAMESPACE.REGION-1.ts.eventgrid.azure.net",
```

Comment:
is the -1 supposed to be there? can it be a different number?

Reply:
no, this is fixed format